### PR TITLE
API-20647 Removal of validation controller

### PR DESCRIPTION
**Pull request by dmekala-va**
## Summary

- This PR is to remove the openid_auth module logic for validation controller as a part of lighthouse which is not used anymore. 
- Lighthouse Team Pivot!

## Related issue(s)
- department-of-veterans-affairs/va.gov-team#[API-20647]
- https://jira.devops.va.gov/browse/API-20647

## Testing done

- Removal of files and dependencies related to validation_controller. Relevant rspecs also deleted. 

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
expected

### CODE REVIEW

File: .github/CODEOWNERS
Comment (dbrown-va) [inline .github/CODEOWNERS:1793]: Any charon related VCR cassettes should be able to be removed as that validation is no longer occurring.

These were incorrectly associated to lighthosue-banana-peels and should have been lighthouse-pivot codeowners when originally rolled out.

File: config/settings.yml
Comment (dbrown-va) [inline config/settings.yml:808]: The Charon call is specifically for SSOi OAuth validation events, and once TV is removed here these settings are no longer applicable.

We will want corresponding removals in the manifests repo
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/tree/main/apps/vets-api/

**Review by dbrown-va — CHANGES_REQUESTED** [review]
The entire `openid_auth` module is a candidate for deletion here, leaving only the parent `app/controllers/openid_application_controller.rb` functionality.

File: .github/CODEOWNERS
Comment (dmekala-va) [reply .github/CODEOWNERS:1793]: removing the "spec/support/vcr_cassettes/charon" alerts a CODEOWNERS entry error to add an entry for each of the files that were deleted and keeping "spec/support/vcr_cassettes/charon" alerts an error to delete it. Adding an entry for each of the files is the only thing that keeps codeowners happy

File: config/settings.yml
Comment (dmekala-va) [reply config/settings.yml:808]: https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2293
https://github.com/department-of-veterans-affairs/vsp-infra-application-manifests/pull/2294

File: .github/CODEOWNERS
Comment (dbrown-va) [inline .github/CODEOWNERS:1794]: Charon specific cassettes should either be removed or marked under lighthouse-pivot as it is auth specific.

**Review by dbrown-va — APPROVED** [review]
Confirmed all external traffic to the opendid_auth module controller has been transitioned.

Remaining traffic to parent controller is deprecated and soon to be removed as well.

**Comment by rmtolmach** [comment]
There are still some files left in the openid_auth module that aren't being removed in this PR. Is that intentional?

**Review by rmtolmach — APPROVED** [review]

**Comment by rmtolmach** [comment]
`/lib/lighthouse` files are incorrectly owned by lighthouse-banana-peels. I will merge without that approval.

**Pull request merged**
